rpc/src/v1/helpers/dispatch.rs

```diff
@@ -449,28 +477,61 @@ fn sign_transaction(
 #[derive(Debug, Clone, Copy)]
 enum ProspectiveSignerState {
 	TryProspectiveSign,
+	PostSign,
+	WaitForPostSign,
 	WaitForNonce,
-	Finish,
 }
 
-struct ProspectiveSigner {
+struct ProspectiveSigner<P: PostSign> {
 	accounts: Arc<AccountProvider>,
 	filled: FilledTransactionRequest,
 	chain_id: Option<u64>,
 	reserved: nonce::Reserved,
 	password: SignWith,
 	state: ProspectiveSignerState,
-	prospective: Option<Result<WithToken<SignedTransaction>>>,
+	prospective: Option<WithToken<SignedTransaction>>,
 	ready: Option<nonce::Ready>,
+	post_sign: Option<P>,
+	post_sign_future: Option<<<P as PostSign>::Out as IntoFuture>::Future>
+}
+
+/// action to execute after signing
+/// e.g importing a transaction into the chain
+pub trait PostSign: Send + Sync {
```

Comment:
Is `Sync` required here? It can be called only once anyway.

Reply:
Yeah, I think you are right